### PR TITLE
[FAQ-Plus] Tunning sample app behavior to align with original app 

### PR DESCRIPTION
**Pull request by dilin-MS**
1. Validate user input title
2. User notification card show "Unassigned" at first.
3. Convert time to user local time
4. Fix SME card type: 1) Chat with <expert> 2) "Unassigned" at first.

### CODE REVIEW

File: faq-plus/bot/cards/smeTicketCard.ts [hunk 1]
```diff
@@ -36,14 +36,11 @@ import { ChangeTicketStatusPayload } from "../models/changeTicketStatusPayload";
  * @param localTimestamp Local timestamp of the user activity.
  * @returns Returns the attachment that will be sent in a message.
  */
-export function getSmeTicketCard(
-  ticket: TicketEntity,
-  localTimestamp?: Date
-): Attachment {
+export function getSmeTicketCard(ticket: TicketEntity): Attachment {
```
Comment (blackchoey) [inline faq-plus/bot/cards/smeTicketCard.ts:39]: why remove the `localTimestamp` parameter?

File: faq-plus/bot/cards/userNotificationCard.ts [hunk 2]
```diff
@@ -33,13 +33,12 @@ import { TicketState } from "../models/ticketState";
  */
 export function getUserNotificationCard(
   ticket: TicketEntity,
-  message: string,
-  activityLocalTimestamp: Date
+  message: string
```
Comment (blackchoey) [inline faq-plus/bot/cards/userNotificationCard.ts:36]: also why remove the 'activityLocalTimestamp` parameter here?

File: faq-plus/bot/common/adaptiveHelper.ts [hunk 3]
```diff
@@ -115,15 +115,10 @@ export function getTicketDisplayStatusForSme(ticket: TicketEntity): string {
...
  * @returns A datetime string.
  */
-export function getFormattedDateInUserTimeZone(
-  dateTime: Date,
-  userLocalTime?: Date
-): string {
-  // todo
-  return dateTime.toString();
+export function getFormattedDateInUserTimeZone(dateTime: Date): string {
+  return dateTime.toLocaleDateString();
```
Comment (blackchoey) [inline faq-plus/bot/common/adaptiveHelper.ts:121]: The implementation of existing sample app converted the time to user's time zone, we may need similar logic here.

File: faq-plus/bot/cards/userNotificationCard.ts [hunk 4]
```diff
@@ -114,19 +106,19 @@ function buildFactSet(
...
-            displayText: TextString.AskAnExpertDisplayText,
-            text: Constants.AskAnExpertSubmitText,
-          }
-        } as AskAnExpertCardPayload;
-      
-        return [action];
-    }
-
-    return [];
+  if (ticket.Status === TicketState.Closed) {
```
Comment (wenytang-ms) [inline faq-plus/bot/cards/userNotificationCard.ts:109]: this function seem build an action not a list of actions, this name is confuse.

File: faq-plus/bot/cards/smeTicketCard.ts (hunk 1 above)
Comment (dilin-MS) [reply faq-plus/bot/cards/smeTicketCard.ts:39]: No need of it now.

File: faq-plus/bot/cards/userNotificationCard.ts (hunk 2 above)
Comment (dilin-MS) [reply faq-plus/bot/cards/userNotificationCard.ts:36]: No need of it now.

File: faq-plus/bot/common/adaptiveHelper.ts (hunk 3 above)
Comment (dilin-MS) [reply faq-plus/bot/common/adaptiveHelper.ts:121]: Yes, seems the `toLocaleDateString()` does it.

File: faq-plus/bot/cards/userNotificationCard.ts (hunk 4 above)
Comment (dilin-MS) [reply faq-plus/bot/cards/userNotificationCard.ts:109]: The function does build a list of action with list length being 1.

File: faq-plus/bot/common/adaptiveHelper.ts (hunk 3 above)
Comment (blackchoey) [reply faq-plus/bot/common/adaptiveHelper.ts:121]: The code runs on server side, how does it know user's time zone if we don't tell it based on bot context?

File: faq-plus/bot/cards/smeTicketCard.ts (hunk 1 above)
Comment (dilin-MS) [reply faq-plus/bot/cards/smeTicketCard.ts]: fixed.

File: faq-plus/bot/common/adaptiveHelper.ts (hunk 3 above)
Comment (dilin-MS) [reply faq-plus/bot/common/adaptiveHelper.ts:126]: fixed.

File: faq-plus/bot/common/adaptiveHelper.ts [hunk 5]
```diff
@@ -115,9 +115,14 @@ export function getTicketDisplayStatusForSme(ticket: TicketEntity): string {
 /**
  * Returns a string that will display the given date and time in the user's local time zone, when placed in an adaptive card.
  * @param dateTime The date and time to format.
+ * @param userLocalTime The sender's local time, as determined by the local timestamp of the activity.
  * @returns A datetime string.
  */
-export function getFormattedDateInUserTimeZone(dateTime: Date): string {
+export function getFormattedDateInUserTimeZone(dateTime: Date, userLocalTime?: Date): string {
+  if (userLocalTime) {
+    let offset = userLocalTime.getTimezoneOffset();
```
Comment (blackchoey) [inline faq-plus/bot/common/adaptiveHelper.ts:123]: const?

**Review by blackchoey — APPROVED** [review]

**Pull request merged**